luaPackages.luv: 1.48.0-2 -> 1.50.0-1

stasjok opened the pull request:
<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->
* Instead of pinned generated package, luv was migrated to manually crafted package.
* Tests were enabled.
* Added passthru.tests.
* Added passthru.updateScript.
* luv dynamically linked to libuv.
* Updated luv to 1.50.0-1.

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

stasjok committed:
luaPackages.luv: 1.48.0-2 -> 1.50.0-1

teto commented:
could we interest you in running the tests as well ? the Makefile has a "test" target that calls lua scripts that I assume dont have many dependencies (but maybe some submodules).

stasjok commented:
> could we interest you in running the tests as well ? the Makefile has a "test" target that calls lua scripts that I assume dont have many dependencies (but maybe some submodules).

I managed to run the tests by changing src temporary:

```diff
diff --git a/pkgs/development/lua-modules/generated-packages.nix b/pkgs/development/lua-modules/generated-packages.nix
index cefa8c7e67cd..a8de4e6ad357 100644
--- a/pkgs/development/lua-modules/generated-packages.nix
+++ b/pkgs/development/lua-modules/generated-packages.nix
@@ -2526,9 +2526,11 @@ buildLuarocksPackage {
     url    = "mirror://luarocks/luv-1.50.0-1.rockspec";
     sha256 = "01i6hs3nllbdwlwcfgjdbxmzgww2yk6a8bn0s5q4lkv67dx89g90";
   }).outPath;
-  src = fetchurl {
-    url    = "https://github.com/luvit/luv/releases/download/1.50.0-1/luv-1.50.0-1.tar.gz";
-    sha256 = "0d5wnn35asqg3ixmyqq80s7ibhbdzl9kxn7dy9a1v64w9l1c6ryq";
+  src = fetchgit {
+    url = "https://github.com/luvit/luv.git";
+    rev = "1.50.0-1";
+    fetchSubmodules = true;
+    hash = "sha256-PS3+qpELpX0tr7UqrlnE4NYScJb50j+9J4fbH9CTr/s=";
   };
 
   disabled = luaOlder "5.1";
diff --git a/pkgs/development/lua-modules/overrides.nix b/pkgs/development/lua-modules/overrides.nix
index 5bf89a654b84..e0955dd1450e 100644
--- a/pkgs/development/lua-modules/overrides.nix
+++ b/pkgs/development/lua-modules/overrides.nix
@@ -867,10 +867,12 @@ in
 
     # we unset the LUA_PATH since the hook erases the interpreter defaults (To fix)
     # tests is not run since they are not part of the tarball anymore
-    preCheck = ''
+    installCheckPhase = ''
       unset LUA_PATH
       rm tests/test-{dns,thread}.lua
+      lua tests/run.lua
     '';
+    doInstallCheck = true;
   });
 
   lyaml = prev.lyaml.overrideAttrs (oa: {
```

Based on comments and commit history, tests were disabled because there is no tests directory in the published tarball. Here is a result:

```
luajit2.1-luv> Running phase: installCheckPhase
luajit2.1-luv> cmd:     lua
luajit2.1-luv> cwd:     /build/luv
luajit2.1-luv> 1..135
luajit2.1-luv>   "in async notify callback"
luajit2.1-luv>   "a"    true    250
luajit2.1-luv> ok 1 async - test pass async between threads
luajit2.1-luv>   "timeout"      uv_timer_t: 0x00496160
luajit2.1-luv>   "closed"       uv_timer_t: 0x00496160
luajit2.1-luv> ok 2 callbacks - luv_handle_t: function
luajit2.1-luv>   "timeout"      {  }    uv_timer_t: 0x00496160
luajit2.1-luv>   "closed"       {  }    uv_timer_t: 0x00496160
luajit2.1-luv> ok 3 callbacks - luv_handle_t: callable table
luajit2.1-luv> ok 4 callbacks - luv_req_t: function
luajit2.1-luv> ok 5 callbacks - luv_req_t: callable table
luajit2.1-luv>   skipped
luajit2.1-luv> ok 6 conversions - basic 64bit conversions
luajit2.1-luv>   Skipping! This test is expected to fail on Lua 5.1 and LuaJIT.
luajit2.1-luv> ok 7 coroutines-require - callback should be in main thread
luajit2.1-luv>   begin wait
luajit2.1-luv>   end wait
luajit2.1-luv> ok 8 coroutines - coroutines
luajit2.1-luv>   { fd = 11 }
luajit2.1-luv>   {
luajit2.1-luv>     stat = {
luajit2.1-luv>       dev = 49,
luajit2.1-luv>       mode = 33188,
luajit2.1-luv>       nlink = 1,
luajit2.1-luv>       uid = 1000,
luajit2.1-luv>       gid = 100,
luajit2.1-luv>       rdev = 0,
luajit2.1-luv>       ino = 40373566,
luajit2.1-luv>       blksize = 4096,
luajit2.1-luv>       blocks = 16,
luajit2.1-luv>       size = 6060,
luajit2.1-luv>       gen = 0,
luajit2.1-luv>       atime = table: 0x7ffff7be17c0,
luajit2.1-luv>       mtime = table: 0x7ffff7bf6c88,
luajit2.1-luv>       ctime = table: 0x7ffff7beec68,
luajit2.1-luv>       birthtime = table: 0x7ffff7ba9800,
luajit2.1-luv>       type = "file",
luajit2.1-luv>       flags = 0
luajit2.1-luv>     }
luajit2.1-luv>   }
luajit2.1-luv> ok 9 fs - read a file sync
luajit2.1-luv> ok 10 fs - read a file sync in chunks
luajit2.1-luv>   { fd = 11 }
luajit2.1-luv>   {
luajit2.1-luv>     stat = {
luajit2.1-luv>       dev = 49,
luajit2.1-luv>       mode = 33188,
luajit2.1-luv>       nlink = 1,
luajit2.1-luv>       uid = 1000,
luajit2.1-luv>       gid = 100,
luajit2.1-luv>       rdev = 0,
luajit2.1-luv>       ino = 40373566,
luajit2.1-luv>       blksize = 4096,
luajit2.1-luv>       blocks = 16,
luajit2.1-luv>       flags = 0,
luajit2.1-luv>       gen = 0,
luajit2.1-luv>       atime = table: 0x7ffff7be17c0,
luajit2.1-luv>       mtime = table: 0x7ffff7bf4c18,
luajit2.1-luv>       ctime = table: 0x7ffff7ba9800,
luajit2.1-luv>       birthtime = table: 0x7ffff7be8758,
luajit2.1-luv>       type = "file",
luajit2.1-luv>       size = 6060
luajit2.1-luv>     }
luajit2.1-luv>   }
luajit2.1-luv>   { chunk = 6060 }
luajit2.1-luv> ok 11 fs - read a file async
luajit2.1-luv> ok 12 fs - fs.write
luajit2.1-luv> ok 13 fs - fs.write data refs
luajit2.1-luv> ok 14 fs - fs.stat sync
luajit2.1-luv> ok 15 fs - fs.stat async
luajit2.1-luv>   { err = "ENOENT: no such file or directory: BAD_FILE!", code = "ENOENT" }
luajit2.1-luv> ok 16 fs - fs.stat sync error
luajit2.1-luv>   { err = "ENOENT: no such file or directory: BAD_FILE@" }
luajit2.1-luv> ok 17 fs - fs.stat async error
luajit2.1-luv>   { name = ".ci", ftype = "directory" }
luajit2.1-luv>   { name = ".clang-format", ftype = "file" }
luajit2.1-luv>   { name = ".clangd", ftype = "file" }
luajit2.1-luv>   { name = ".editorconfig", ftype = "file" }
luajit2.1-luv>   { name = ".github", ftype = "directory" }
luajit2.1-luv>   { name = ".gitignore", ftype = "file" }
luajit2.1-luv>   { name = ".gitmodules", ftype = "file" }
luajit2.1-luv>   { name = "CMakeLists.txt", ftype = "file" }
luajit2.1-luv>   { name = "LICENSE.txt", ftype = "file" }
luajit2.1-luv>   { name = "Makefile", ftype = "file" }
luajit2.1-luv>   { name = "README.md", ftype = "file" }
luajit2.1-luv>   { name = "appveyor.yml", ftype = "file" }
luajit2.1-luv>   { name = "build.luarocks", ftype = "directory" }
luajit2.1-luv>   { name = "cmake", ftype = "directory" }
luajit2.1-luv>   { name = "deps", ftype = "directory" }
luajit2.1-luv>   { name = "docs.md", ftype = "file" }
luajit2.1-luv>   { name = "examples", ftype = "directory" }
luajit2.1-luv>   { name = "lib", ftype = "directory" }
luajit2.1-luv>   { name = "libluv.pc.in", ftype = "file" }
luajit2.1-luv>   { name = "luv-scm-0.rockspec", ftype = "file" }
luajit2.1-luv>   { name = "msvcbuild.bat", ftype = "file" }
luajit2.1-luv>   { name = "rockspecs", ftype = "directory" }
luajit2.1-luv>   { name = "src", ftype = "directory" }
luajit2.1-luv>   { name = "tests", ftype = "directory" }
luajit2.1-luv> ok 18 fs - fs.scandir
luajit2.1-luv>   { err = "ENOENT: no such file or directory: BAD_FILE!", code = "ENOENT" }
luajit2.1-luv> ok 19 fs - fs.scandir sync error
luajit2.1-luv>   { err = "ENOENT: no such file or directory: BAD_FILE!" }
luajit2.1-luv> ok 20 fs - fs.scandir async error
luajit2.1-luv>   { name = ".ci", ftype = "directory" }
luajit2.1-luv>   { name = ".clang-format", ftype = "file" }
luajit2.1-luv>   { name = ".clangd", ftype = "file" }
luajit2.1-luv>   { name = ".editorconfig", ftype = "file" }
luajit2.1-luv>   { name = ".github", ftype = "directory" }
luajit2.1-luv>   { name = ".gitignore", ftype = "file" }
luajit2.1-luv>   { name = ".gitmodules", ftype = "file" }
luajit2.1-luv>   { name = "CMakeLists.txt", ftype = "file" }
luajit2.1-luv>   { name = "LICENSE.txt", ftype = "file" }
luajit2.1-luv>   { name = "Makefile", ftype = "file" }
luajit2.1-luv>   { name = "README.md", ftype = "file" }
luajit2.1-luv>   { name = "appveyor.yml", ftype = "file" }
luajit2.1-luv>   { name = "build.luarocks", ftype = "directory" }
luajit2.1-luv>   { name = "cmake", ftype = "directory" }
luajit2.1-luv>   { name = "deps", ftype = "directory" }
luajit2.1-luv>   { name = "docs.md", ftype = "file" }
luajit2.1-luv>   { name = "examples", ftype = "directory" }
luajit2.1-luv>   { name = "lib", ftype = "directory" }
luajit2.1-luv>   { name = "libluv.pc.in", ftype = "file" }
luajit2.1-luv>   { name = "luv-scm-0.rockspec", ftype = "file" }
luajit2.1-luv>   { name = "msvcbuild.bat", ftype = "file" }
luajit2.1-luv>   { name = "rockspecs", ftype = "directory" }
luajit2.1-luv>   { name = "src", ftype = "directory" }
luajit2.1-luv>   { name = "tests", ftype = "directory" }
luajit2.1-luv> ok 21 fs - fs.scandir async
luajit2.1-luv> ok 22 fs - fs.scandir with no iteration
luajit2.1-luv> ok 23 fs - fs.scandir given to new_work
luajit2.1-luv>   "/build/luv"
luajit2.1-luv>   "/build/luv"
luajit2.1-luv> ok 24 fs - fs.realpath
luajit2.1-luv> ok 25 fs - fs.copyfile
luajit2.1-luv>   { { name = "LICENSE.txt", type = "file" } }
luajit2.1-luv>   { { name = "README.md", type = "file" } }
luajit2.1-luv>   { { name = "docs.md", type = "file" } }
luajit2.1-luv>   { { name = ".clangd", type = "file" } }
luajit2.1-luv>   { { name = ".clang-format", type = "file" } }
luajit2.1-luv>   { { name = ".editorconfig", type = "file" } }
luajit2.1-luv>   { { name = ".gitignore", type = "file" } }
luajit2.1-luv>   { { name = ".gitmodules", type = "file" } }
luajit2.1-luv>   { { name = "Makefile", type = "file" } }
luajit2.1-luv>   { { name = "appveyor.yml", type = "file" } }
luajit2.1-luv>   { { name = "libluv.pc.in", type = "file" } }
luajit2.1-luv>   { { name = "luv-scm-0.rockspec", type = "file" } }
luajit2.1-luv>   { { name = "msvcbuild.bat", type = "file" } }
luajit2.1-luv>   { { name = "CMakeLists.txt", type = "file" } }
luajit2.1-luv>   { { name = ".ci", type = "directory" } }
luajit2.1-luv>   { { name = ".github", type = "directory" } }
luajit2.1-luv>   { { name = "cmake", type = "directory" } }
luajit2.1-luv>   { { name = "deps", type = "directory" } }
luajit2.1-luv>   { { name = "examples", type = "directory" } }
luajit2.1-luv>   { { name = "lib", type = "directory" } }
luajit2.1-luv>   { { name = "rockspecs", type = "directory" } }
luajit2.1-luv>   { { name = "src", type = "directory" } }
luajit2.1-luv>   { { name = "tests", type = "directory" } }
luajit2.1-luv>   { { name = "build.luarocks", type = "directory" } }
luajit2.1-luv> ok 26 fs - fs.{open,read,close}dir object sync #1
luajit2.1-luv>   { { name = "LICENSE.txt", type = "file" } }
luajit2.1-luv>   { { name = "README.md", type = "file" } }
luajit2.1-luv>   { { name = "docs.md", type = "file" } }
luajit2.1-luv>   { { name = ".clangd", type = "file" } }
luajit2.1-luv>   { { name = ".clang-format", type = "file" } }
luajit2.1-luv>   { { name = ".editorconfig", type = "file" } }
luajit2.1-luv>   { { name = ".gitignore", type = "file" } }
luajit2.1-luv>   { { name = ".gitmodules", type = "file" } }
luajit2.1-luv>   { { name = "Makefile", type = "file" } }
luajit2.1-luv>   { { name = "appveyor.yml", type = "file" } }
luajit2.1-luv>   { { name = "libluv.pc.in", type = "file" } }
luajit2.1-luv>   { { name = "luv-scm-0.rockspec", type = "file" } }
luajit2.1-luv>   { { name = "msvcbuild.bat", type = "file" } }
luajit2.1-luv>   { { name = "CMakeLists.txt", type = "file" } }
luajit2.1-luv>   { { name = ".ci", type = "directory" } }
luajit2.1-luv>   { { name = ".github", type = "directory" } }
luajit2.1-luv>   { { name = "cmake", type = "directory" } }
luajit2.1-luv>   { { name = "deps", type = "directory" } }
luajit2.1-luv>   { { name = "examples", type = "directory" } }
luajit2.1-luv>   { { name = "lib", type = "directory" } }
luajit2.1-luv>   { { name = "rockspecs", type = "directory" } }
luajit2.1-luv>   { { name = "src", type = "directory" } }
luajit2.1-luv>   { { name = "tests", type = "directory" } }
luajit2.1-luv>   { { name = "build.luarocks", type = "directory" } }
luajit2.1-luv>   uv_dir_t: 0x7ffff7bc5f88       closed
luajit2.1-luv> ok 27 fs - fs.{open,read,close}dir object sync #2
luajit2.1-luv>   { { name = "LICENSE.txt", type = "file" } }
luajit2.1-luv>   { { name = "README.md", type = "file" } }
luajit2.1-luv>   { { name = "docs.md", type = "file" } }
luajit2.1-luv>   { { name = ".clangd", type = "file" } }
luajit2.1-luv>   { { name = ".clang-format", type = "file" } }
luajit2.1-luv>   { { name = ".editorconfig", type = "file" } }
luajit2.1-luv>   { { name = ".gitignore", type = "file" } }
luajit2.1-luv>   { { name = ".gitmodules", type = "file" } }
luajit2.1-luv>   { { name = "Makefile", type = "file" } }
luajit2.1-luv>   { { name = "appveyor.yml", type = "file" } }
luajit2.1-luv>   { { name = "libluv.pc.in", type = "file" } }
luajit2.1-luv>   { { name = "luv-scm-0.rockspec", type = "file" } }
luajit2.1-luv>   { { name = "msvcbuild.bat", type = "file" } }
luajit2.1-luv>   { { name = "CMakeLists.txt", type = "file" } }
luajit2.1-luv>   { { name = ".ci", type = "directory" } }
luajit2.1-luv>   { { name = ".github", type = "directory" } }
luajit2.1-luv>   { { name = "cmake", type = "directory" } }
luajit2.1-luv>   { { name = "deps", type = "directory" } }
luajit2.1-luv>   { { name = "examples", type = "directory" } }
luajit2.1-luv>   { { name = "lib", type = "directory" } }
luajit2.1-luv>   { { name = "rockspecs", type = "directory" } }
luajit2.1-luv>   { { name = "src", type = "directory" } }
luajit2.1-luv>   { { name = "tests", type = "directory" } }
luajit2.1-luv>   { { name = "build.luarocks", type = "directory" } }
luajit2.1-luv> ok 28 fs - fs.{open,read,close}dir sync one entry
luajit2.1-luv>   {
luajit2.1-luv>     { name = "LICENSE.txt", type = "file" },
luajit2.1-luv>     { name = "README.md", type = "file" },
luajit2.1-luv>     { name = "docs.md", type = "file" },
luajit2.1-luv>     { name = ".clangd", type = "file" },
luajit2.1-luv>     { name = ".clang-format", type = "file" },
luajit2.1-luv>     { name = ".editorconfig", type = "file" },
luajit2.1-luv>     { name = ".gitignore", type = "file" },
luajit2.1-luv>     { name = ".gitmodules", type = "file" },
luajit2.1-luv>     { name = "Makefile", type = "file" },
luajit2.1-luv>     { name = "appveyor.yml", type = "file" },
luajit2.1-luv>     { name = "libluv.pc.in", type = "file" },
luajit2.1-luv>     { name = "luv-scm-0.rockspec", type = "file" },
luajit2.1-luv>     { name = "msvcbuild.bat", type = "file" },
luajit2.1-luv>     { name = "CMakeLists.txt", type = "file" },
luajit2.1-luv>     { name = ".ci", type = "directory" },
luajit2.1-luv>     { name = ".github", type = "directory" },
luajit2.1-luv>     { name = "cmake", type = "directory" },
luajit2.1-luv>     { name = "deps", type = "directory" },
luajit2.1-luv>     { name = "examples", type = "directory" },
luajit2.1-luv>     { name = "lib", type = "directory" },
luajit2.1-luv>     { name = "rockspecs", type = "directory" },
luajit2.1-luv>     { name = "src", type = "directory" },
luajit2.1-luv>     { name = "tests", type = "directory" },
luajit2.1-luv>     { name = "build.luarocks", type = "directory" }
luajit2.1-luv>   }
luajit2.1-luv> ok 29 fs - fs.{open,read,close}dir sync more entry
luajit2.1-luv>   {
luajit2.1-luv>     { name = "LICENSE.txt", type = "file" },
luajit2.1-luv>     { name = "README.md", type = "file" },
luajit2.1-luv>     { name = "docs.md", type = "file" },
luajit2.1-luv>     { name = ".clangd", type = "file" },
luajit2.1-luv>     { name = ".clang-format", type = "file" },
luajit2.1-luv>     { name = ".editorconfig", type = "file" },
luajit2.1-luv>     { name = ".gitignore", type = "file" },
luajit2.1-luv>     { name = ".gitmodules", type = "file" },
luajit2.1-luv>     { name = "Makefile", type = "file" },
luajit2.1-luv>     { name = "appveyor.yml", type = "file" },
luajit2.1-luv>     { name = "libluv.pc.in", type = "file" },
luajit2.1-luv>     { name = "luv-scm-0.rockspec", type = "file" },
luajit2.1-luv>     { name = "msvcbuild.bat", type = "file" },
luajit2.1-luv>     { name = "CMakeLists.txt", type = "file" },
luajit2.1-luv>     { name = ".ci", type = "directory" },
luajit2.1-luv>     { name = ".github", type = "directory" },
luajit2.1-luv>     { name = "cmake", type = "directory" },
luajit2.1-luv>     { name = "deps", type = "directory" },
luajit2.1-luv>     { name = "examples", type = "directory" },
luajit2.1-luv>     { name = "lib", type = "directory" },
luajit2.1-luv>     { name = "rockspecs", type = "directory" },
luajit2.1-luv>     { name = "src", type = "directory" },
luajit2.1-luv>     { name = "tests", type = "directory" },
luajit2.1-luv>     { name = "build.luarocks", type = "directory" }
luajit2.1-luv>   }
luajit2.1-luv> ok 30 fs - fs.{open,read,close}dir with more entry
luajit2.1-luv> ok 31 fs - fs.opendir and fs.closedir in a loop
luajit2.1-luv>   {
luajit2.1-luv>     { name = "LICENSE.txt", type = "file" },
luajit2.1-luv>     { name = "README.md", type = "file" },
luajit2.1-luv>     { name = "docs.md", type = "file" },
luajit2.1-luv>     { name = ".clangd", type = "file" },
luajit2.1-luv>     { name = ".clang-format", type = "file" },
luajit2.1-luv>     { name = ".editorconfig", type = "file" },
luajit2.1-luv>     { name = ".gitignore", type = "file" },
luajit2.1-luv>     { name = ".gitmodules", type = "file" },
luajit2.1-luv>     { name = "Makefile", type = "file" },
luajit2.1-luv>     { name = "appveyor.yml", type = "file" },
luajit2.1-luv>     { name = "libluv.pc.in", type = "file" },
luajit2.1-luv>     { name = "luv-scm-0.rockspec", type = "file" },
luajit2.1-luv>     { name = "msvcbuild.bat", type = "file" },
luajit2.1-luv>     { name = "CMakeLists.txt", type = "file" },
luajit2.1-luv>     { name = ".ci", type = "directory" },
luajit2.1-luv>     { name = ".github", type = "directory" },
luajit2.1-luv>     { name = "cmake", type = "directory" },
luajit2.1-luv>     { name = "deps", type = "directory" },
luajit2.1-luv>     { name = "examples", type = "directory" },
luajit2.1-luv>     { name = "lib", type = "directory" },
luajit2.1-luv>     { name = "rockspecs", type = "directory" },
luajit2.1-luv>     { name = "src", type = "directory" },
luajit2.1-luv>     { name = "tests", type = "directory" },
luajit2.1-luv>     { name = "build.luarocks", type = "directory" }
luajit2.1-luv>   }
luajit2.1-luv> ok 32 fs - fs.{open,read,close}dir ref check
luajit2.1-luv>   { ffree = 0, bavail = 11106024, type = 2435016766, bsize = 4096, blocks = 41811707, bfree = 14782575, 
files = 0 }
luajit2.1-luv> ok 33 fs - fs.statfs sync
luajit2.1-luv>   { bfree = 14782575, bavail = 11106024, ffree = 0, type = 2435016766, files = 0, bsize = 4096, blocks =
 41811707 }
luajit2.1-luv> ok 34 fs - fs.statfs async
luajit2.1-luv>   { err = "ENOENT: no such file or directory: BAD_FILE!", code = "ENOENT" }
luajit2.1-luv> ok 35 fs - fs.statfs sync error
luajit2.1-luv>   { err = "ENOENT: no such file or directory: BAD_FILE@" }
luajit2.1-luv> ok 36 fs - fs.statfs async error
luajit2.1-luv> ok 37 fs - fs.mkdtemp async
luajit2.1-luv> ok 38 fs - fs.mkdtemp sync
luajit2.1-luv> ok 39 fs - fs.mkdtemp async error
luajit2.1-luv> ok 40 fs - fs.mkdtemp sync error
luajit2.1-luv> ok 41 fs - fs.mkstemp async
luajit2.1-luv> ok 42 fs - fs.mkstemp sync
luajit2.1-luv> ok 43 fs - fs.mkstemp async error
luajit2.1-luv> ok 44 fs - fs.mkstemp sync error
luajit2.1-luv>   "fs_rename"    nil     "EISDIR: illegal operation on a directory: _test_ -> _testdir_" "EISDIR"
luajit2.1-luv>   "fs_link"      nil     "EEXIST: file already exists: _test_ -> _testdir_"      "EEXIST"
luajit2.1-luv>   "fs_symlink"   nil     "EEXIST: file already exists: _test_ -> _testdir_"      "EEXIST"
luajit2.1-luv>   "fs_copyfile"  nil     "EISDIR: illegal operation on a directory: _test_ -> _testdir_" "EISDIR"
luajit2.1-luv> ok 45 fs - errors with dest paths
luajit2.1-luv> ok 46 handle - get type
luajit2.1-luv> ok 47 loop - uv.loop_mode
luajit2.1-luv>   "stdout"       {  }
luajit2.1-luv>   #437   0       0
luajit2.1-luv> ok 48 loop - issue #437, crash without uv.run
luajit2.1-luv>   "stdout"       {  }
luajit2.1-luv>   #599   5       0
luajit2.1-luv> ok 49 loop - issue #599, crash during calling os.exit
luajit2.1-luv>   { loop_count = 62, events = 30, events_waiting = 0 }
luajit2.1-luv>   { loop_count = 63, events = 30, events_waiting = 0 }
luajit2.1-luv> ok 50 metrics - idle time
luajit2.1-luv>   "stdio fd types"       { [0] = "file", [1] = "tty", [2] = "tty" }
luajit2.1-luv> ok 51 misc - uv.guess_handle
luajit2.1-luv>   { version = 78336, version_string = "1.50.0" }
luajit2.1-luv> ok 52 misc - uv.version and uv.version_string
luajit2.1-luv>   { rss = 4788224, total = 16687112192, constrained = 0, available = 12455006208, free = 12455006208 }
luajit2.1-luv> ok 53 misc - memory size
luajit2.1-luv>   { uptime = 18790265.57 }
luajit2.1-luv> ok 54 misc - uv.uptime
luajit2.1-luv>   {
luajit2.1-luv>     isrss = 0,
luajit2.1-luv>     minflt = 812,
luajit2.1-luv>     majflt = 0,
luajit2.1-luv>     nswap = 0,
luajit2.1-luv>     inblock = 0,
luajit2.1-luv>     oublock = 256,
luajit2.1-luv>     msgsnd = 0,
luajit2.1-luv>     msgrcv = 0,
luajit2.1-luv>     nsignals = 0,
luajit2.1-luv>     nvcsw = 45,
luajit2.1-luv>     nivcsw = 6,
luajit2.1-luv>     utime = { usec = 638461, sec = 0 },
luajit2.1-luv>     stime = { usec = 11954, sec = 0 },
luajit2.1-luv>     maxrss = 4676,
luajit2.1-luv>     ixrss = 0,
luajit2.1-luv>     idrss = 0
luajit2.1-luv>   }
luajit2.1-luv> ok 55 misc - uv.getrusage
luajit2.1-luv>   {
luajit2.1-luv>     stime = { usec = 7938, sec = 0 },
luajit2.1-luv>     maxrss = 4676,
luajit2.1-luv>     ixrss = 0,
luajit2.1-luv>     idrss = 0,
luajit2.1-luv>     isrss = 0,
luajit2.1-luv>     minflt = 729,
luajit2.1-luv>     majflt = 0,
luajit2.1-luv>     nswap = 0,
luajit2.1-luv>     inblock = 0,
luajit2.1-luv>     oublock = 208,
luajit2.1-luv>     nivcsw = 3,
luajit2.1-luv>     nvcsw = 15,
luajit2.1-luv>     msgsnd = 0,
luajit2.1-luv>     msgrcv = 0,
luajit2.1-luv>     nsignals = 0,
luajit2.1-luv>     utime = { usec = 640615, sec = 0 }
luajit2.1-luv>   }
luajit2.1-luv> ok 56 misc - uv.getrusage_thread
luajit2.1-luv>   8
luajit2.1-luv> ok 57 misc - uv.available_parallelism
luajit2.1-luv>   {
luajit2.1-luv>     { model = "AMD Ryzen Threadripper 2950X 16-Core Processor", speed = 0, times = table: 0x7ffff7be7480
 },
luajit2.1-luv>     { model = "AMD Ryzen Threadripper 2950X 16-Core Processor", speed = 0, times = table: 0x7ffff7be1ad0
 },
luajit2.1-luv>     { model = "AMD Ryzen Threadripper 2950X 16-Core Processor", speed = 0, times = table: 0x7ffff7bb3958
 },
luajit2.1-luv>     { model = "AMD Ryzen Threadripper 2950X 16-Core Processor", speed = 0, times = table: 0x7ffff7ba9800
 },
luajit2.1-luv>     { model = "AMD Ryzen Threadripper 2950X 16-Core Processor", speed = 0, times = table: 0x7ffff7be17c0
 },
luajit2.1-luv>     { model = "AMD Ryzen Threadripper 2950X 16-Core Processor", speed = 0, times = table: 0x7ffff7ba6d40
 },
luajit2.1-luv>     { model = "AMD Ryzen Threadripper 2950X 16-Core Processor", speed = 0, times = table: 0x7ffff7bf6bc0
 },
luajit2.1-luv>     { model = "AMD Ryzen Threadripper 2950X 16-Core Processor", speed = 0, times = table: 0x7ffff7c90b08
 }
luajit2.1-luv>   }
luajit2.1-luv> ok 58 misc - uv.cpu_info
luajit2.1-luv>   "lo"   {
luajit2.1-luv>     {
luajit2.1-luv>       ip = "127.0.0.1",
luajit2.1-luv>       internal = true,
luajit2.1-luv>       mac = "\0\0\0\0\0\0",
luajit2.1-luv>       family = "inet",
luajit2.1-luv>       netmask = "255.0.0.0"
luajit2.1-luv>     },
luajit2.1-luv>     {
luajit2.1-luv>       ip = "::1",
luajit2.1-luv>       internal = true,
luajit2.1-luv>       mac = "\0\0\0\0\0\0",
luajit2.1-luv>       family = "inet6",
luajit2.1-luv>       netmask = "ffff:ffff:ffff:ffff:ffff:ffff:ffff:ffff"
luajit2.1-luv>     }
luajit2.1-luv>   }
luajit2.1-luv> ok 59 misc - uv.interface_addresses
luajit2.1-luv>   { 0.78, 0.42, 0.51 }
luajit2.1-luv> ok 60 misc - uv.loadavg
luajit2.1-luv>   "/nix/store/8ynfggrvlipzmzm80irw97zhns0y2srp-luajit-2.1.1741730670/bin/luajit-2.1.1741730670"
luajit2.1-luv> ok 61 misc - uv.exepath
luajit2.1-luv>   "/homeless-shelter"
luajit2.1-luv> ok 62 misc - uv.os_homedir
luajit2.1-luv>   "/build"
luajit2.1-luv> ok 63 misc - uv.os_tmpdir
luajit2.1-luv>   { homedir = "/build", username = "nixbld", uid = 1000, gid = 100, shell = "/noshell" }
luajit2.1-luv> ok 64 misc - uv.os_get_passwd
luajit2.1-luv>   "/build/luv"
luajit2.1-luv>   "/"
luajit2.1-luv> ok 65 misc - uv.cwd and uv.chdir
luajit2.1-luv>   1.8790265582138e+16
luajit2.1-luv> ok 66 misc - uv.hrtime
luajit2.1-luv> ok 67 misc - uv.getpid
luajit2.1-luv>   {
luajit2.1-luv>     release = "5.10.0-32-amd64",
luajit2.1-luv>     version = "#1 SMP Debian 5.10.223-1 (2024-08-10)",
luajit2.1-luv>     sysname = "Linux",
luajit2.1-luv>     machine = "x86_64"
luajit2.1-luv>   }
luajit2.1-luv> ok 68 misc - uv.os_uname
luajit2.1-luv>           os.time        1743193977
luajit2.1-luv>   uv.gettimeofday        1743193977.000992
luajit2.1-luv> ok 69 misc - uv.gettimeofday
luajit2.1-luv> ok 70 misc - uv.os_environ
luajit2.1-luv> ok 71 misc - uv.sleep
luajit2.1-luv> ok 72 misc - uv.random async
luajit2.1-luv> ok 73 misc - uv.random sync
luajit2.1-luv> ok 74 misc - uv.random errors
luajit2.1-luv> ok 75 misc - uv errno
luajit2.1-luv> ok 76 misc - uv constants
luajit2.1-luv>   1024   nil
luajit2.1-luv> ok 77 misc - uv.cpumask_size
luajit2.1-luv>   "monotonic"    { nsec = 588894169, sec = 18790266 }    nil
luajit2.1-luv>   "realtime"     { nsec = 588915148, sec = 18790266 }    nil
luajit2.1-luv> ok 78 misc - uv.clock_gettime
luajit2.1-luv> ok 79 misc - uv.wtf8 and utf16 conversion
luajit2.1-luv> ok 80 misc - uv.wtf8<->utf16 unpaired surrogate
luajit2.1-luv> ok 81 misc - uv.wtf8<->utf16 empty strings
luajit2.1-luv> ok 82 pipe - pipe chmod
luajit2.1-luv> ok 83 pipe - pipe ping pong
luajit2.1-luv> ok 84 pipe - pipe close fd
luajit2.1-luv> ok 85 pipe - pipe getsockname abstract
luajit2.1-luv>   "prepare"      uv_prepare_t: 0x004974c0
luajit2.1-luv> ok 86 prepare-check-idle-async - simple prepare
luajit2.1-luv>   "check"        uv_check_t: 0x004974c0
luajit2.1-luv>   "timeout"      uv_timer_t: 0x00499750
luajit2.1-luv> ok 87 prepare-check-idle-async - simple check
luajit2.1-luv>   "idle" uv_idle_t: 0x004974c0
luajit2.1-luv> ok 88 prepare-check-idle-async - simple idle
luajit2.1-luv> ok 89 prepare-check-idle-async - simple async
luajit2.1-luv> ok 90 process - test disable_stdio_inheritance
luajit2.1-luv>   { pid = 801, handle = uv_process_t: 0x00495bd0 }
luajit2.1-luv>   "stdout"       { chunk = "Hello World\n" }
luajit2.1-luv>   "exit" { code = 0, signal = 0 }
luajit2.1-luv> ok 91 process - process stdout
luajit2.1-luv>   { pid = 802, handle = uv_process_t: 0x00495bd0 }
luajit2.1-luv>   "exit" uv_process_t: 0x00495bd0        { signal = 2, status = 0 }
luajit2.1-luv> ok 92 process - spawn and kill by pid
luajit2.1-luv>   { pid = 803, handle = uv_process_t: 0x00495bd0 }
luajit2.1-luv>   "exit" uv_process_t: 0x00495bd0        { signal = 15, status = 0 }
luajit2.1-luv> ok 93 process - spawn and kill by handle
luajit2.1-luv> ok 94 process - invalid command
luajit2.1-luv>   { pid = 805, handle = uv_process_t: 0x00495bd0 }
luajit2.1-luv>   "stdout"       { chunk = "Hello World" }
luajit2.1-luv>   "exit" { code = 0, signal = 0 }
luajit2.1-luv> ok 95 process - process stdio
luajit2.1-luv> ok 96 process - get pid
luajit2.1-luv> ok 97 req - cancel
luajit2.1-luv> ok 98 req - get type
luajit2.1-luv>   Sending child SIGINT
luajit2.1-luv>   "sigint\n"
luajit2.1-luv>   "exit" { code = 0, pid = 807, signal = 0 }
luajit2.1-luv> ok 99 signal - Catch SIGINT
luajit2.1-luv>   { server = uv_tcp_t: 0x00499650, address = { port = 39879, ip = "127.0.0.1", family = "inet" } }
luajit2.1-luv>   { client = uv_tcp_t: 0x00499ac0, req = uv_connect_t: 0x7ffff7babdd0 }
luajit2.1-luv>   "server on connection" uv_tcp_t: 0x00499650
luajit2.1-luv>   "client on connect"    uv_tcp_t: 0x00499ac0    nil
luajit2.1-luv>   "client on shutdown"   uv_tcp_t: 0x00499ac0    nil
luajit2.1-luv>   "client on close"      uv_tcp_t: 0x00499ac0
luajit2.1-luv> ok 100 tcp - basic tcp server and client (ipv4)
luajit2.1-luv>   { server = uv_tcp_t: 0x00499650, address = { port = 37543, ip = "::1", family = "inet6" } }
luajit2.1-luv>   { client = uv_tcp_t: 0x00499ac0, req = uv_connect_t: 0x7ffff7bca720 }
luajit2.1-luv>   "server on connection" uv_tcp_t: 0x00499650
luajit2.1-luv>   "client on connect"    uv_tcp_t: 0x00499ac0    nil
luajit2.1-luv>   "client on shutdown"   uv_tcp_t: 0x00499ac0    nil
luajit2.1-luv>   "client on close"      uv_tcp_t: 0x00499ac0
luajit2.1-luv> ok 101 tcp - basic tcp server and client (ipv6)
luajit2.1-luv>   { server = uv_tcp_t: 0x00499650, address = { port = 44225, ip = "127.0.0.1", family = "inet" } }
luajit2.1-luv>   { socket = uv_tcp_t: 0x00499ac0, req = uv_write_t: 0x7ffff7be1348 }
luajit2.1-luv>   "client onwrite"       uv_tcp_t: 0x00499ac0    nil
luajit2.1-luv>   "server read"  { data = "Hello" }
luajit2.1-luv>   "client read"  { data = "Hello" }
luajit2.1-luv>   "server read"  {  }
luajit2.1-luv> ok 102 tcp - tcp echo server and client
luajit2.1-luv>   { server = uv_tcp_t: 0x00499650, address = { port = 37183, ip = "127.0.0.1", family = "inet" } }
luajit2.1-luv>   { socket = uv_tcp_t: 0x00499bc0, req = uv_write_t: 0x7ffff7c871a0 }
luajit2.1-luv>   "client onwrite"       uv_tcp_t: 0x00499bc0    nil
luajit2.1-luv>   "server read"  { data = "Hello" }
luajit2.1-luv>   "client read"  { data = "Hello" }
luajit2.1-luv>   "server read"  {  }
luajit2.1-luv> ok 103 tcp - tcp echo server and client with methods
luajit2.1-luv>   "./tests/test-tcp.lua:140: Invalid IP address or port [127.0.0.100005:1000]"
luajit2.1-luv> ok 104 tcp - tcp invalid ip address
luajit2.1-luv>   { server = uv_tcp_t: 0x00499650, address = { port = 39131, ip = "127.0.0.1", family = "inet" } }
luajit2.1-luv>   { socket = uv_tcp_t: 0x00499ac0, req = uv_write_t: 0x7ffff7c871a0 }
luajit2.1-luv>   "client onwrite"       uv_tcp_t: 0x00499ac0    nil
luajit2.1-luv>   "server read"  { data = "Hello" }
luajit2.1-luv>   "client read"  { data = "Hello" }
luajit2.1-luv>   "server read"  { err = "ECONNRESET" }
luajit2.1-luv> ok 105 tcp - tcp close reset client
luajit2.1-luv>   { server = uv_tcp_t: 0x00499650, address = { port = 42219, ip = "127.0.0.1", family = "inet" } }
luajit2.1-luv>   { socket = uv_tcp_t: 0x00499bc0, req = uv_write_t: 0x7ffff7be1348 }
luajit2.1-luv>   "client onwrite"       uv_tcp_t: 0x00499bc0    nil
luajit2.1-luv>   "server read"  { data = "Hello" }
luajit2.1-luv>   "client read"  { data = "Hello" }
luajit2.1-luv>   "server read"  {  }
luajit2.1-luv> ok 106 tcp - tcp close reset after shutdown
luajit2.1-luv>   { server = uv_tcp_t: 0x00499650, address = { port = 43939, ip = "127.0.0.1", family = "inet" } }
luajit2.1-luv>   { socket = uv_tcp_t: 0x00499ac0, req = uv_write_t: 0x7ffff7be21b8 }
luajit2.1-luv>   "client onwrite"       uv_tcp_t: 0x00499ac0    nil
luajit2.1-luv>   "server read"  { data = "Hello" }
luajit2.1-luv>   "client read"  { data = "Hello" }
luajit2.1-luv>   "client read"  {  }
luajit2.1-luv> ok 107 tcp - tcp close reset accepted
luajit2.1-luv>   { server = uv_tcp_t: 0x00499650, address = { port = 42513, ip = "127.0.0.1", family = "inet" } }
luajit2.1-luv>   { socket = uv_tcp_t: 0x00499ac0, req = uv_write_t: 0x7ffff7c871a0 }
luajit2.1-luv>   "client onwrite"       uv_tcp_t: 0x00499ac0    nil
luajit2.1-luv>   "server read"  { data = "Hello" }
luajit2.1-luv>   "client read"  { data = "Hello" }
luajit2.1-luv>   "server read"  {  }
luajit2.1-luv> ok 108 tcp - tcp close reset accepted after shutdown
luajit2.1-luv> ok 109 tcp - socketpair ping pong
luajit2.1-luv>   "timeout"      uv_timer_t: 0x00499750
luajit2.1-luv>   "closed"       uv_timer_t: 0x00499750
luajit2.1-luv> ok 110 timer - simple timeout
luajit2.1-luv>   "interval"     uv_timer_t: 0x00499750
luajit2.1-luv>   "interval"     uv_timer_t: 0x00499750
luajit2.1-luv>   "interval"     uv_timer_t: 0x00499750
luajit2.1-luv>   "closed"       uv_timer_t: 0x00499750
luajit2.1-luv> ok 111 timer - simple interval
luajit2.1-luv>   "interval"     uv_timer_t: 0x00499cc0
luajit2.1-luv>   "timeout"      uv_timer_t: 0x00499750
luajit2.1-luv>   "interval"     uv_timer_t: 0x00499cc0
luajit2.1-luv> ok 112 timer - timeout with interval
luajit2.1-luv>   "interval"     uv_timer_t: 0x00499cc0  0
luajit2.1-luv>   "interval"     uv_timer_t: 0x00499cc0  8
luajit2.1-luv>   "interval"     uv_timer_t: 0x00499cc0  4
luajit2.1-luv>   "interval"     uv_timer_t: 0x00499cc0  2
luajit2.1-luv> ok 113 timer - shrinking interval
luajit2.1-luv>   "interval"     uv_timer_t: 0x00499cc0  0
luajit2.1-luv>   "interval"     uv_timer_t: 0x00499cc0  8
luajit2.1-luv>   "interval"     uv_timer_t: 0x00499cc0  4
luajit2.1-luv>   "interval"     uv_timer_t: 0x00499cc0  2
luajit2.1-luv> ok 114 timer - shrinking interval using methods
luajit2.1-luv> ok 115 timer - timer init
luajit2.1-luv> ok 116 timer - timer huge timeout
luajit2.1-luv>   ./tests/test-tty.lua:19: bad argument #1 to 'is_readable' (Expected uv_stream userdata)
luajit2.1-luv>   stack traceback:
luajit2.1-luv>          [C]: in function 'is_readable'
luajit2.1-luv>          ./tests/test-tty.lua:19: in function 'fn'
luajit2.1-luv>          ./lib/tap.lua:59: in function <./lib/tap.lua:48>
luajit2.1-luv>          [C]: in function 'xpcall'
luajit2.1-luv>          ./lib/tap.lua:48: in function 'run'
luajit2.1-luv>          ./lib/tap.lua:146: in function 'tap'
luajit2.1-luv>          tests/run.lua:23: in main chunk
luajit2.1-luv>          [C]: at 0x004063b0
luajit2.1-luv> not ok 117 tty - tty normal
luajit2.1-luv> ok 118 tty - tty pty
luajit2.1-luv>   Skip, open /dev/tty fail
luajit2.1-luv> ok 119 tty - tty device
luajit2.1-luv>   { server = uv_udp_t: 0x0049a3d0 }
luajit2.1-luv>   { client = uv_udp_t: 0x0049a4b0, req = 0 }
luajit2.1-luv>   "client on send"       uv_udp_t: 0x0049a4b0    nil
luajit2.1-luv>   "server on recv"       uv_udp_t: 0x0049a3d0    "PING"  { port = 58059, ip = "127.0.0.1", family = "ine
t" }     {  }
luajit2.1-luv>   "server on close"      uv_udp_t: 0x0049a3d0
luajit2.1-luv>   "client on close"      uv_udp_t: 0x0049a4b0
luajit2.1-luv> ok 120 udp - basic udp server and client (ipv4)
luajit2.1-luv>   { server = uv_udp_t: 0x0049a4b0 }
luajit2.1-luv>   { client = uv_udp_t: 0x0049a3d0, req = 0 }
luajit2.1-luv>   "client on send"       uv_udp_t: 0x0049a3d0    nil
luajit2.1-luv>   "server on recv"       uv_udp_t: 0x0049a4b0    "PING"  { port = 44651, ip = "127.0.0.1", family = "ine
t" }     {  }
luajit2.1-luv>   "server on close"      uv_udp_t: 0x0049a4b0
luajit2.1-luv>   "client on close"      uv_udp_t: 0x0049a3d0
luajit2.1-luv> ok 121 udp - basic udp send from table
luajit2.1-luv>   { server = uv_udp_t: 0x0049a3d0 }
luajit2.1-luv>   { client = uv_udp_t: 0x0049a4b0, req = 0 }
luajit2.1-luv>   "client on send"       uv_udp_t: 0x0049a4b0    nil
luajit2.1-luv>   "server on recv"       uv_udp_t: 0x0049a3d0    "PING"  { port = 38474, ip = "::1", family = "inet6" } 
 {  }
luajit2.1-luv>   "server on close"      uv_udp_t: 0x0049a3d0
luajit2.1-luv>   "client on close"      uv_udp_t: 0x0049a4b0
luajit2.1-luv> ok 122 udp - basic udp server and client (ipv6)
luajit2.1-luv>   ./tests/test-udp.lua:85: bad argument #3 to 'udp_send' (Host must be string or nil)
luajit2.1-luv>   ./tests/test-udp.lua:87: bad argument #4 to 'udp_send' (Port must be number or nil)
luajit2.1-luv>   ./tests/test-udp.lua:89: bad argument #3 to 'udp_send' (Both host and port must be nil if one is nil)
luajit2.1-luv>   ./tests/test-udp.lua:91: bad argument #4 to 'udp_send' (Both host and port must be nil if one is nil)
luajit2.1-luv> ok 123 udp - udp send args
luajit2.1-luv>   { server = uv_udp_t: 0x0049a4b0 }
luajit2.1-luv>   { port = 9123, ip = "127.0.0.1", family = "inet" }
luajit2.1-luv>   "server on recv"       uv_udp_t: 0x0049a4b0    "PING"  { port = 37508, ip = "127.0.0.1", family = "ine
t" }     {  }
luajit2.1-luv>   "server on recv"       uv_udp_t: 0x0049a4b0    "PING"  { port = 41261, ip = "127.0.0.1", family = "ine
t" }     {  }
luajit2.1-luv>   "server on recv"       uv_udp_t: 0x0049a4b0    "PING"  { port = 41261, ip = "127.0.0.1", family = "ine
t" }     {  }
luajit2.1-luv>   "server on recv"       uv_udp_t: 0x0049a4b0    nil     nil     {  }
luajit2.1-luv>   "server on recv"       uv_udp_t: 0x0049a4b0    "PING"  { port = 51096, ip = "127.0.0.1", family = "ine
t" }     {  }
luajit2.1-luv>   "client on close"      uv_udp_t: 0x0049a3d0
luajit2.1-luv>   "server on close"      uv_udp_t: 0x0049a4b0
luajit2.1-luv> ok 124 udp - udp connect
luajit2.1-luv>   no external ipv4 interface, skipping
luajit2.1-luv> ok 125 udp - udp multicast join ipv4
luajit2.1-luv>   no external ipv6 interface, skipping
luajit2.1-luv> ok 126 udp - udp multicast join ipv6
luajit2.1-luv>   "PING" { port = 48325, ip = "127.0.0.1", family = "inet" }
luajit2.1-luv>   "PING" { port = 48325, ip = "127.0.0.1", family = "inet" }
luajit2.1-luv>   "PING" { port = 48325, ip = "127.0.0.1", family = "inet" }
luajit2.1-luv>   "PING" { port = 48325, ip = "127.0.0.1", family = "inet" }
luajit2.1-luv>   "PING" { port = 48325, ip = "127.0.0.1", family = "inet" }
luajit2.1-luv>   "PING" { port = 48325, ip = "127.0.0.1", family = "inet" }
luajit2.1-luv>   "PING" { port = 48325, ip = "127.0.0.1", family = "inet" }
luajit2.1-luv>   "PING" { port = 48325, ip = "127.0.0.1", family = "inet" }
luajit2.1-luv> ok 127 udp - udp recvmmsg
luajit2.1-luv>   sent 5 out of 5, expecting 4 to be received
luajit2.1-luv>   "PING" { port = 9125, ip = "127.0.0.1", family = "inet" }
luajit2.1-luv>   "PING" { port = 9125, ip = "127.0.0.1", family = "inet" }
luajit2.1-luv>   "PING" { port = 9125, ip = "127.0.0.1", family = "inet" }
luajit2.1-luv>   "PING" { port = 9125, ip = "127.0.0.1", family = "inet" }
luajit2.1-luv> ok 128 udp - udp try_send2 not connected
luajit2.1-luv>   sent 5 out of 5, expecting 5 to be received
luajit2.1-luv>   "PING" { port = 56564, ip = "127.0.0.1", family = "inet" }
luajit2.1-luv>   "PING" { port = 56564, ip = "127.0.0.1", family = "inet" }
luajit2.1-luv>   "PING" { port = 56564, ip = "127.0.0.1", family = "inet" }
luajit2.1-luv>   "PING" { port = 56564, ip = "127.0.0.1", family = "inet" }
luajit2.1-luv>   "PING" { port = 56564, ip = "127.0.0.1", family = "inet" }
luajit2.1-luv> ok 129 udp - udp try_send2 connected
luajit2.1-luv> ok 130 work - test threadpool with return none
luajit2.1-luv>   "Please be patient, the test cost a lots of time"
luajit2.1-luv>   "run 10%"
luajit2.1-luv>   "run 20%"
luajit2.1-luv>   "run 30%"
luajit2.1-luv>   "run 40%"
luajit2.1-luv>   "run 50%"
luajit2.1-luv>   "run 60%"
luajit2.1-luv>   "run 70%"
luajit2.1-luv>   "run 80%"
luajit2.1-luv>   "run 90%"
luajit2.1-luv>   "run 100%"
luajit2.1-luv> ok 131 work - test threadpool
luajit2.1-luv>   "in async notify callback"
luajit2.1-luv>   "a"    true    250
luajit2.1-luv>   2      4       "uv_thread_t: 0x7ffff7ba36c0"   "hello"
luajit2.1-luv>   uv_thread_t: 0x7ffff7ba36c0    finish  hello
luajit2.1-luv> ok 132 work - test threadpool with async
luajit2.1-luv> 1
luajit2.1-luv> 2
luajit2.1-luv> 3
luajit2.1-luv> 4
luajit2.1-luv> 5
luajit2.1-luv> 6
luajit2.1-luv> 7
luajit2.1-luv> 8
luajit2.1-luv> ok 133 work - test threadpool with coro
luajit2.1-luv> ok 134 work - test threadpool with invalid argument
luajit2.1-luv> Uncaught Error: Error: thread arg not support type 'table' at 1
luajit2.1-luv> stack traceback:
luajit2.1-luv>  [C]: at 0x7ffff7c35510
luajit2.1-luv> ok 135 work - test threadpool with invalid return value
luajit2.1-luv> #1 failed test
error: builder for '/nix/store/lfisag4ib98wvy494h1sh7q5a5k4bi85-luajit2.1-luv-1.50.0-1.drv' failed with exit code 255;
       last 25 log lines:
       >   "run 70%"
       >   "run 80%"
       >   "run 90%"
       >   "run 100%"
       > ok 131 work - test threadpool
       >   "in async notify callback"
       >   "a"       true    250
       >   2        4       "uv_thread_t: 0x7ffff7ba36c0"   "hello"
       >   uv_thread_t: 0x7ffff7ba36c0  finish  hello
       > ok 132 work - test threadpool with async
       > 1
       > 2
       > 3
       > 4
       > 5
       > 6
       > 7
       > 8
       > ok 133 work - test threadpool with coro
       > ok 134 work - test threadpool with invalid argument
       > Uncaught Error: Error: thread arg not support type 'table' at 1
       > stack traceback:
       >       [C]: at 0x7ffff7c35510
       > ok 135 work - test threadpool with invalid return value
       > #1 failed test
       For full logs, run 'nix log /nix/store/lfisag4ib98wvy494h1sh7q5a5k4bi85-luajit2.1-luv-1.50.0-1.drv'.
```

One test failed, but it doesn't look like a fail to me, because «Uncaught Error» is exactly what is expected by the test: https://github.com/luvit/luv/blob/ec4185f98209d5e5cf96885ab126d4c2a26dfaa9/tests/test-work.lua#L123. Maybe tests were ran not exactly as they should be.

stasjok commented:
> One test failed, but it doesn't look like a fail to me, because «Uncaught Error» is exactly what is expected by the test: https://github.com/luvit/luv/blob/ec4185f98209d5e5cf96885ab126d4c2a26dfaa9/tests/test-work.lua#L123. Maybe tests were ran not exactly as they should be.

No, It's wrong. A different test failed: **not ok** 117 tty - tty normal. I think tty isn't available in chrooted build environment. If I remove test-tty.lua, then all tests passed. If I run tests without build environment (in nix develop shell), they are all passed.

stasjok commented:
Also I noticed that luv is not linked to libuv, but compiled from a vendored libuv (deps/libuv directory). libluv is linked correctly. I don't know how fix it or whether it need to be fixed.

teto commented:
It's not a problem if we dont run 100% of the tests. In the sandbox it's hard anyway but it's helpful to detect some big breakage so feel free to remove the tests that dont pass, ideally with a comment. If we build from a git checkout rather than tarball it's fine.

If it compiles against a vendored library that should be fixed to use nixpkgs' libuv, there is no reason here I think of using a vendored lib. Thanks for trying <3

stasjok commented:
> It's not a problem if we dont run 100% of the tests. In the sandbox it's hard anyway but it's helpful to detect some big breakage so feel free to remove the tests that dont pass, ideally with a comment. If we build from a git checkout rather than tarball it's fine.

This package is generated from luarocks. Tarball from luarocks doesn't have tests in it. So this is possible only if we package this manually (not using luarocks-packages-updater). Is this what you want me to do?

stasjok committed:
luaPackages.luv: use manual packaging

* add tests
* add update script
* build with shared libluv

stasjok commented:
> If it compiles against a vendored library that should be fixed to use nixpkgs' libuv, there is no reason here I think of using a vendored lib. Thanks for trying <3

I moved it to a separate package. To fix a shared library it need a hack with patching rockspec, otherwise variables are not passed to cmake. Also I've added update script.

stasjok commented:
Will it get merged? I'd like to get it into 25.05.

teto commented:
sry it slipped through my notifs. I'll have a look

stasjok commented:
@teto

teto commented:
this looks ok and this is the version advised for neovim 0.11. I've tested on top of master. @GaetanLepage feel free to merge if that's ok for you too

GaetanLepage reviewed (approved):
Thanks @teto and @stasjok

GaetanLepage commented:
Apparently, this broke `luv` on darwin...
